### PR TITLE
Add support for the MotorGo Mini board

### PR DESCRIPTION
**Pull request by SwapnilPande**
Add support for MotorGo Mini 1 motor controller with ESP32-S3.

-----------
## Description of Change
Adds a new variant for the MotorGo Mini with an onboard ESP32-S3. Updated boards.txt and created a new variants directory.

## Tests scenarios
Tested the new variant with the Arduino-ESP32Core 2.0.14 running on the MotorGo Mini 1 board. 

## Related links
[MotorGo Website](https://motorgo.net/)

### CODE REVIEW

**Comment by lucasssvaz** [comment]
IMO it would be better to add it to the 3.0.0 version (master branch).

**Comment by P-R-O-C-H-Y** [comment]
@SwapnilPande Overall looks good, but as Lucas mentioned, to have the board available for new releases you should target master branch. Also as we are planning 2.0.15 release, you can leave this Rand open a new one targeting the master.

**Comment by SwapnilPande** [comment]
Thanks for the quick review! Wanted to make sure I target the 3.0 and 2.0 releases, I'll submit a PR targeting master today as well.

**Comment by SwapnilPande** [comment]
Went ahead and rebased on the latest commit as well :)

File: variants/motorgo_mini_1/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#define USB_VID 0x303A
+#define USB_PID 0x1001
+
+#define NUM_DIGITAL_PINS 6
+#define NUM_ANALOG_INPUTS 1
+
+#define analogInputToDigitalPin(p) \
+  (((p) < 20) ? (analogChannelToDigitalPin(p)) : -1)
+#define digitalPinToInterrupt(p) (((p) < 48) ? (p) : -1)
+#define digitalPinHasPWM(p) (p < 46)
```
Comment (lucasssvaz) [inline variants/motorgo_mini_1/pins_arduino.h:17]: These can be deleted as they are globally defined.

```suggestion
```

Comment (SwapnilPande) [reply variants/motorgo_mini_1/pins_arduino.h:17]: I included these based on [this PR](https://github.com/espressif/arduino-esp32/pull/9263), which says that they do need to be included for 2.x releases.

Are `NUM_DIGITAL_PINS` and `NUM_ANALOG INPUTS` used internally? I changed the values to match the number of pins exposed to users of our board, but I realize now that they might need to match the number available on the microcontroller.

Comment (lucasssvaz) [reply variants/motorgo_mini_1/pins_arduino.h:17]: Ah yes, my bad. For 2.x it is still required.

Comment (P-R-O-C-H-Y) [reply variants/motorgo_mini_1/pins_arduino.h:17]: @SwapnilPande They are not used internally. You can leave it as it is :) This way most of boards are done in 2.X

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Pull request merged**